src/posit/connect/content.py

```diff
@@ -137,12 +140,231 @@ class ContentItem(Resource):
         Tags associated with the content item.
     """
 
+    # CRUD Methods
+
+    def delete(self) -> None:
+        """Delete the content item."""
+        path = f"v1/content/{self.guid}"
+        url = urls.append(self.config.url, path)
+        self.session.delete(url)
+
+    def deploy(self) -> tasks.Task:
+        """Deploy the content.
+
+        Spawns an asynchronous task, which activates the latest bundle.
+
+        Returns
+        -------
+        tasks.Task
+            The task for the deployment.
+
+        Examples
+        --------
+        >>> task = content.deploy()
+        >>> task.wait_for()
+        None
+        """
+        path = f"v1/content/{self.guid}/deploy"
+        url = urls.append(self.config.url, path)
+        response = self.session.post(url, json={"bundle_id": None})
+        result = response.json()
+        ts = tasks.Tasks(self.config, self.session)
+        return ts.get(result["task_id"])
+
+    def refresh(self) -> Task | None:
+        """Trigger a content refresh.
+
+        Submit a refresh request to the server for the content. After submission, the server executes an asynchronous process to refresh the content. This is useful when content is dependent on external information, such as a dataset.
+
+        See Also
+        --------
+        restart
+
+        Notes
+        -----
+        This method is identical to `restart` and exists to provide contextual clarity. Both methods produce identical results. When working with documents, natural language prefers "refresh this content" instead of "restart this content" since documents do not require a system process. When writing software that operates on multiple types of content (e.g., applications, documents, scripts, etc.), you may use either 'refresh' or 'restart' to achive the same result.
+
+        Examples
+        --------
+        >>> refresh()
+        """
+        return self._re_whatever()
+
+    def restart(self) -> Task | None:
+        """Initiate a content restart.
+
+        Sends a restart request to the server for the content. Once submitted, the server performs an asynchronous process to restart the content. This is particularly useful when the content relies on external information loaded into application memory, such as datasets. Additionally, restarting can help clear memory leaks or reduce excessive memory usage that might build up over time.
+
+        See Also
+        --------
+        refresh
+
+        Notes
+        -----
+        This method is identical to `refresh` and exists to provide contextual clarity. Both methods produce identical results. When working with applications, natural language prefers "restart this content" instead of "refresh this content" since applications require a system process. When writing software that operates on multiple types of content (e.g., applications, documents, scripts, etc.), you may use either 'restart' or 'refresh' to achieve the same result.
```

Comment:
1. I don't know that "restart" and "refresh" should be identical. I might think you have `restart()`, which works on interactive content (and errors on static), and `render()`, which works on static (and errors on interactive), and then `refresh()` is the generic. Then you have

```python
def refresh(self, **kwargs):
    if self.is_interactive():
        self.restart()
    else:
        self.render(**kwargs) # for variants/params
```

That way, if you want to be strict about what you're intending (restart or render), you can.

2. If you want to proceed in the current direction, you don't need _re_whatever(), just define one and have the other call it, you don't need them both to call some third function.

Reply:
After discussing, we decided to implement `restart` and `render' and provide a guide error message if the content type does not support the invoked method.